lib/Onyx.js

```diff
@@ -804,7 +807,14 @@ function notifyCollectionSubscribersOnNextTick(key, value) {
 function remove(key) {
     cache.drop(key);
     notifySubscribersOnNextTick(key, null);
-    return Storage.removeItem(key);
+
+    const writePromise = keyOperations[key] || Promise.resolve();
```

Comment:
Since now many functions are using this logic, and it might not be clear to other what we're doing here, it would be best to extract it to a function like `queueModification` .
This would make it clearer and the function itself might provide further documentation on its usage / use case 

```js
function queueModification(key, runModification) {
  const currentTask = keyOperations[key] || Promise.resolve();
  keyOperations[key] = currentTask.then(runModification);
  return keyOperations[key];
}
```

```js
function remove(key) {
    cache.drop(key);
    notifySubscribersOnNextTick(key, null);

   const nextModification = () => 
       Storage.removeItem(key)
              .catch(error => Logger.logAlert(`An error occurred while removing key: ${key}, Error: ${error}`)));

    return queueModification(key, nextModification);
}
```

Later `queueModification` can serve as optimization hatch, where it can dedupe or skip redundant ops